Sources/MessagingPushFCM/MessagingPushFCM.swift

```diff
@@ -90,7 +90,7 @@ public class MessagingPushFCM: MessagingPushFCMInstance {
         _ request: UNNotificationRequest,
         withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
     ) -> Bool {
-        messagingPush.didReceive(request, withContentHandler: contentHandler)
+        (messagingPush as MessagingPushInstance).didReceive(request, withContentHandler: contentHandler)
```

Comment:
I don't understand why this works? 

Should we be doing this `(messagingPush as MessagingPushInstance)` for the other functions in this file such as tracking metric, deleting device token, etc? 

Also adding a code comment explaining why this works would be appreciated to explain why this fixes the bug.